src/meta/processors/admin/Balancer.h

```diff
@@ -0,0 +1,96 @@
+/* Copyright (c) 2019 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#ifndef META_ADMIN_BALANCERR_H_
+#define META_ADMIN_BALANCERR_H_
+
+#include <gtest/gtest_prod.h>
+#include <folly/executors/CPUThreadPoolExecutor.h>
+#include "time/TimeUtils.h"
+#include "kvstore/KVStore.h"
+#include "network/NetworkUtils.h"
+#include "meta/processors/admin/AdminClient.h"
+#include "meta/processors/admin/BalanceTask.h"
+#include "meta/processors/admin/BalancePlan.h"
+
+namespace nebula {
+namespace meta {
+/**
+1. Balance will generate balance plan according to current active hosts and parts allocation
+2. For the plan, we hope after moving the least parts , it will reach a reasonable state.
```

Comment:
Could be better if we put the rational of `Why we want to to move the LEAST instead of the MAXIMAL parts` here?
Intuitionally，BALANCE means to move data from the MORE side to the LESS side. Why do it counter-intuitively?
IMO, we should doc the following(at least):

- What we are balancing? 
- What's the criteria of  balance/imbalance?
- How can we admin it?
- How can we tune it?

Reply:
Maybe @dangleptr means "moving the minimum parts in quanity" here.

Reply:
Yes.  The principle is we reach balanced state with the least part movements